### PR TITLE
Test that app has ota and ota has listeners for OTA operations

### PR DESCRIPTION
**Pull request by systemcrash**
Apparently in some situations they're absent. Let's guard against them.

Closes issue #225

### CODE REVIEW

File: custom_components/zha_toolkit/ota.py [hunk 1]
```diff
@@ -158,28 +158,34 @@ async def download_sonoff_ota(listener, ota_dir):
...
-            url = getattr(image, "url", None)
-            LOGGER.error("Try getting %r, %r, %r", image_key, url, image)
-            try:
-                img = await app.ota.get_ota_image(
-                    image_key.manufacturer_id, image_key.image_type, model=None
-                )
-                LOGGER.info("Got image %r", getattr(img, "header", None))
-            except Exception as e:
-                LOGGER.error("%r while getting %r - %s", e, image_key, url)
+    if hasattr(app, 'ota') and hasattr(app.ota, '_listeners')::
```
Comment (frli4797) [inline custom_components/zha_toolkit/ota.py:161]: Syntax error '::'

File: custom_components/zha_toolkit/ota.py [hunk 2]
```diff
@@ -158,28 +158,34 @@ async def download_sonoff_ota(listener, ota_dir):
...
+    else:
+        LOGGER.warning("Could not get ota object for download_zigpy_ota, try again")
 
 
 async def ota_update_images(
     app, listener, ieee, cmd, data, service, params, event_data
 ):
-    for _, (ota, _) in app.ota._listeners.items():
-        await ota.refresh_firmware_list()
+    if hasattr(app, 'ota') and hasattr(app.ota, '_listeners')::
```
Comment (frli4797) [inline custom_components/zha_toolkit/ota.py:184]: Syntax error on '::'

**Review by frli4797 — CHANGES_REQUESTED** [review]
This commit seems to have to have two syntax errors with double colons on conditional statements. (Typo?) I've tried you PR with just removing these, which worked fine.

File: custom_components/zha_toolkit/ota.py (hunk 1 above)
Comment (systemcrash) [reply custom_components/zha_toolkit/ota.py]: yeah, typo

**Comment by mdeweerd** [comment]
I've been quite busy for a while, I have just integrated this and making a release with the updates.
Thank you.

**Pull request merged**